api/middleware/synchronize.js

```diff
@@ -65,13 +70,30 @@ const synchronizeAll = getSpecifics =>
  *      - action: (optional) the name of the action being
  *          performed; sent to the client in the case of
  *          a client error (e.g., validation failure)
+ * @param {object} events Event handlers.  Must be async/return
+ *    a promise.  If an event handler throws, the request will
+ *    terminate with a 500 server error unless the thrown error
+ *    has a statusCode property.
+ * @param {SyncEventHandler} events.afterSync Called after the model has
+ *    been synced.
+ * @param {SyncEventHandler} events.beforeSync Called before the model is
+ *    synced.  A good opportunity to massage the request object.
  */
-const synchronizeSpecific = getModelToSync =>
+const synchronizeSpecific = (getModelToSync, { afterSync, beforeSync } = {}) =>
```

Comment:
Add some lifecycle-type events.  We need this for the APD `put` route in order to update the state model before the response is sent.  I think ideally the callbacks would also happen inside the db transaction, but...  I dunno.  This seems okay?

Reply:
nice, seems legit to me